workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder.md

```diff
@@ -0,0 +1,254 @@
+   > **https://github.com/your-username/my-first-binder**
+
+3) As you type, the webpage generates a link in the "Copy the URL below..." box
+
+   Should look like this:
+   > **https://mybinder.org/v2/gh/your-username/my-first-binder/master**
+
+4) Copy it, open a new browser tab and visit that URL.
+   * You will see a "spinner" as Binder launches the repo.
+
+If everything ran smoothly, you'll see a Jupyter Notebook interface.
+
+#### What's happening in the background?
+
+While you wait, BinderHub (the backend of Binder) is:
+* Fetching your repo from GitHub
+* Analysing the contents
+* Creating a Docker file
+* Launching that Docker image in the Cloud
+* Connecting you to it via your browser
+
+## 3. Run `hello.py`
+
+**TO DO:** :vertical_traffic_light:
+
+1. In the top right corner, click "New" :arrow_right: "Terminal"
+2. In the new tab with the terminal, type `python hello.py` and press return
+
+`Hello from Binder!` should be printed to the terminal.
+
+## 4. Pinning Dependencies
+
+It was easy to get started, but our environment is barebones - let's add a **dependency**!
+
+**TO DO:** :vertical_traffic_light:
+
+1) In your repo, create a file called `requirements.txt`
+2) Add a line that says: `numpy==1.14.5`
+3) Check for typos!
+4) Visit **https://mybinder.org/v2/gh/your-username/my-first-binder/master** again in a new tab
+
+This time, click on "Build Logs" in the big, horizontal, grey bar.
+This will let you watch the progress of your build.
+It's useful when your build fails or something you think _should_ be installed is missing.
+
+#### What's happening in the background?
+
+This time, BinderHub will read `requirements.txt` and install version `1.14.5` of the `numpy` package.
+
+#### More on pinning dependencies
+
+In the above example, we used two equals signs (`==`) to pin the version of `numpy`.
+This tells Binder to install that _specific_ version.
+
+Another way to pin a version number is to use the greater than or equal to sign (`>=`) to allow any version above a particular one to be installed.
+This is useful when you have a lot of dependencies that may have dependencies on each other and allows Binder to find a configuration of your dependencies that do not conflict with one another whilst avoiding any earlier versions which may break or change your code.
+
+Finally, you could not provide a version number at all (just the name of the library/package) and Binder will install the latest version of that package.
+
+## 5. Check the Environment
+
+**TO DO:** :vertical_traffic_light:
+
+1) In the top right corner, click "New" :arrow_right: "Python 3" to open a new notebook
+
+2) Type the following into a new cell:
+   ```python
+   import numpy
+   print(numpy.__version__)
+   numpy.random.randn()
+   ```
+   **Note the two underscores either side of `version`!**
+
+3) Run the cell to see the version number and a random number printed out
+   * Press either SHIFT+RETURN or the "Run" button in the Menu bar
```

Comment:
One last comment - make sure you mention that you can't save this file into the repo from Binder.

Reply:
:+1: noted!